src/shared/telemetry/telemetryUtils.ts

```diff
@@ -85,3 +88,9 @@ export function registerCommand<T>({
         thisArg
     )
 }
+
+function setMetadataIfNotExists(metadata: Map<string, string>, key: string, value: string) {
+    if (!metadata.has(key)) {
```

Comment:
`Map.set` is idempotent, so the only case where this test would matter is if it was previously set to a *different* value. In this case, should we throw?

Reply:
No, we wish to use the existing value if there is one